Sources/NIOInstrumentation/ContextInboundHTTPHandler.swift

```diff
@@ -0,0 +1,27 @@
+import ContextPropagation
+import NIO
+import NIOHTTP1
+
+public final class ContextInboundHTTPHandler: ChannelInboundHandler {
+    public typealias InboundIn = HTTPServerRequestPart
+    public typealias InboundOut = HTTPServerRequestPart
+
+    private let instrumentationMiddleware: InstrumentationMiddleware<HTTPHeaders, HTTPHeaders>
+    private var onContext: ((Context) -> Void)?
```

Comment:
Minor: since the onContext init parameter is not optional this does not have to be either